src/media/track.ts

```diff
@@ -0,0 +1,220 @@
+
+    /**
+     * Emit mute event when the underlying track gets muted.
+     */
+    this.#mediaStreamTrack.onmute = () => {
+      // using arrow function which should bind to this from outer scope track
+      const action = this.#mediaStreamTrack.enabled ? 'muted' : 'unmuted';
+      // TODO:  Move this logic else where
+
+      this.emit('track:mute', {
+        action,
+      });
+    };
+  }
+
+  /**
+   * Get id of this track.
+   *
+   * @returns The id of this track.
+   */
+  get id(): string {
+    return this.#mediaStreamTrack.id;
+  }
+
+  /**
+   * Get muted state of this track.
+   *
+   * @returns The muted state of this track.
+   */
+  get muted(): boolean {
+    return !this.#mediaStreamTrack.enabled;
+  }
+
+  /**
+   * Get current state of this track.
+   *
+   * @returns Current state of this track.
+   */
+  get trackState(): TrackState {
+    return { id: this.id, label: this.label, muted: this.muted };
+  }
+
+  /**
+   * Set the mute state of this track.
+   *
+   * @param muted - True to mute, false to unmute.
+   * @fires Events.Muted
+   */
+  setMuted(muted: boolean): void {
+    // Only change state if it's different, where "enabled" means "unmuted."
+    if (this.#mediaStreamTrack.enabled === muted) {
+      this.#mediaStreamTrack.enabled = !muted;
+      this.emit(Events.Muted, { trackState: this.trackState });
+      logger.log(`Local track ${muted ? 'muted' : 'unmuted'}:`, { trackState: this.trackState });
+    }
+  }
+
+  /**
+   *
+   * This method returns the actual MediaStreamTrack.
+   *
+   * @returns #mediaStreamTrack of type MediaStreamTrack.
+   */
+  getMediaStreamTrack(): MediaStreamTrack {
+    return this.#mediaStreamTrack;
+  }
+
+  /**
+   *
+   * This method returns track after applying effects.
+   *
+   * @returns #mediaStreamTrack of type MediaStreamTrack.
+   */
+  getMediaStreamTrackWithEffects(): MediaStreamTrack {
+    return this.#mediaStreamTrackWithEffect;
+  }
+
+  /**
+   *
+   * This method sets the track after applying effect.
+   *
+   * @param track
+   * @returns #mediaStreamTrack of type MediaStreamTrack.
+   */
+  setMediaStreamTrackWithEffects(track: MediaStreamTrack): void {
+    this.#mediaStreamTrackWithEffect = track;
+    this.emit(Events.UnderlyingTrackChange);
+  }
+
+  /**
+   * Stop this track.
+   *
+   * @fires Events.Ended
+   */
+  stop(): void {
+    this.#mediaStreamTrack.stop();
+    this.emit(Events.Ended, { trackState: this.trackState });
+    logger.log(`Local track stopped:`, { trackState: this.trackState });
+  }
+
+  /**
+   *
+   * @param element - Either pass the ID for the dom element where we need to attach the video tag
+   * pass the video element dom element where we need to attach the stream.
+   */
+
+  /**
+   * @param element
+   */
+  play(element?: string | HTMLElement): void {
+    if (element instanceof HTMLElement) {
+      // eslint-disable-next-line no-param-reassign
+      element.srcObject = new MediaStream([this.#mediaStreamTrack]);
```

Comment:
if we have effects applied, the srcObject should probably use a stream constructed from `this.#mediaStreamTrackWithEffect` track, right? (same in line 216)

Reply:
added to https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-398390